eureka2-client/src/main/java/com/netflix/eureka2/client/channel/InterestChannelInvoker.java

```diff
@@ -28,6 +32,11 @@ public InterestChannelInvoker(InterestChannel delegate, EurekaClientMetricFactor
     public InterestChannelInvoker(InterestChannel delegate, EurekaClientMetricFactory metricFactory, Scheduler scheduler) {
         super(metricFactory.getSerializedTaskInvokerMetrics(InterestChannelInvoker.class), scheduler);
         this.delegate = delegate;
+        if (delegate instanceof Sourced) {
```

Comment:
In actual implementation yes, but this is not enforced by the current InterestChannel contract. Going forward I propose we have all the channel interfaces extend the Sourced interface.